packages/nextra-theme-docs/src/constants.tsx

```diff
@@ -167,14 +191,17 @@ export const DEFAULT_THEME: DocsThemeConfig = {
     loading: function useLoading() {
       const { locale, defaultLocale = DEFAULT_LOCALE } = useRouter()
       const text =
-        (locale && LOADING_LOCALES[locale]) || LOADING_LOCALES[defaultLocale]
-      return <>{text}…</>
+        (locale && LOADING_LOCALES[locale]) ||
+        LOADING_LOCALES[defaultLocale] ||
+        LOADING_LOCALES[DEFAULT_LOCALE]
+      return `${text}…`
     },
     placeholder: function usePlaceholder() {
       const { locale, defaultLocale = DEFAULT_LOCALE } = useRouter()
       const text =
         (locale && PLACEHOLDER_LOCALES[locale]) ||
-        PLACEHOLDER_LOCALES[defaultLocale]
+        PLACEHOLDER_LOCALES[defaultLocale] ||
+        PLACEHOLDER_LOCALES[DEFAULT_LOCALE]
       return `${text}…`
```

Comment:
Avoid displaying `undefined` as text. For example, if  `defaultLocale: ja-JP`, it might display as `undefined...` by default.

Reply:
I think it's better to completely remove default values for other languages different from `en` for i18n properties, to keep it simple and not increase bundle size